app/models/agent.rb

```diff
@@ -466,4 +466,12 @@ def type
       @object.__send__(attr)
     } unless method_defined?(attr)
   }
+
+  def working
+    @object.working?
+  end
+
+  def url
+    Rails.application.routes.url_helpers.agent_url(@object, Rails.application.config.action_mailer.default_url_options)
```

Comment:
`Rails.application.config.action_mailer.default_url_options` feels like a little misleading indirection, does `agent_url` raise when `ENV['DOMAIN']` is not set?

Reply:
It does, saying `ArgumentError: Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true`.